module_generator/README.md

```diff
@@ -0,0 +1,220 @@
+# Module Generator Template
+
+This is the Python module template for [CLI module generator](https://docs.viam.com/cli/#module).
+
+## How to Use
+
+1. Install cookiecutter: `pipx install cookiecutter`. If pipx is not an option, use `python -m pip install --user cookiecutter`.
+2. Call the template: `python3 module_generator/cookiecutter-template`.
+3. Fill out the prompts given (see below).
+4. The necessary files for the created module will be generated in a new directory `module_generator/<MODULE_NAME>`.
+
+NOTE: The imports in `module_generator/<MODULE_NAME>/src/__main__.py` may have to be adjusted. More details below in the example.
+
+## Prompts
+
+To create a module, the generator needs the user to fill out information about the module. The prompts can be seen in `cookiecutter-template/cookiecutter.json`. The public prompts that a user directly affects are as follows:
+
+- `module_name` (str): Set the name of the module.
+- `language` (int): Specify the language for the module being created. Currently, the options are (1) Python.
+- `namespace` (str): Set the name of the namespace or organization ID in which the module is being created.
+- `resource` (int): Select a resource from a numbered list to be added to the module.
+- `model_name` (str): Set the name of the model of the resource.
+- `initialize_workflow` (bool): Initialize a workflow for building the module and other cloud build info.
+
+## Template Structure
+
+When a module is created using the template, there are a couple of different files created or populated. Some files to note:
+
+- `.viam-gen-info` saves the user inputs, the module creation time, and the generator version.
+- `meta.json` saves the `module_name`, `visibility` (default to public), `language`, `generator_version`, `model`, `api`, and `entrypoint` (default to `run.sh`) into a json file.
+- `src/__main__.py` contains all the functions available to the resource.
+
+---
+
+## Module Creation Example
+
+Create a Python module named `rover` in the namespace `test`. Add a vision service model `vizz`. Do not initialize a github workflow.
+
+```python
+% python3 cookiecutter-template
+  [1/6] module_name (MyModule): rover
+  [2/6] Select language
+    1 - python
+    Choose from [1] (1): 1
+  [3/6] namespace (ns): test
+  [4/6] Select resource
+    1 - arm component
+    2 - base component
+    3 - motor component
+    4 - vision service
+    Choose from [1/2/3/4] (1): 4
+  [5/6] model_name (model): vizz
+  [6/6] initialize_workflow (n): n
+```
+
+### Generated Files
+
+`module_generator/rover/.viam-gen-info`
+
+```json
+{
+  "module_name": "rover",
+  "language": "python",
+  "namespace": "test",
+  "resource": "vision service",
+  "model_name": "vizz",
+  "generator_version": "0.14.0",
```

Comment:
(q) Just out of curiosity, where this `generator_version` come from? It looks like we try to use the SDK version and fallback on `0.1.0`, how did we end up with `0.14.0`?